test/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/NRLMSISE00.test.cpp

```diff
@@ -0,0 +1,358 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Core/Containers/Array.hpp>
+#include <OpenSpaceToolkit/Core/Containers/Table.hpp>
+#include <OpenSpaceToolkit/Core/FileSystem/File.hpp>
+#include <OpenSpaceToolkit/Core/FileSystem/Path.hpp>
+#include <OpenSpaceToolkit/Core/Types/Index.hpp>
+#include <OpenSpaceToolkit/Core/Types/Real.hpp>
+#include <OpenSpaceToolkit/Core/Types/Shared.hpp>
+#include <OpenSpaceToolkit/Core/Types/String.hpp>
+
+#include <OpenSpaceToolkit/Physics/Coordinate/Frame.hpp>
+#include <OpenSpaceToolkit/Physics/Coordinate/Position.hpp>
+#include <OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/NRLMSISE00.hpp>
+#include <OpenSpaceToolkit/Physics/Environment/Ephemerides/SPICE.hpp>
+#include <OpenSpaceToolkit/Physics/Environment/Objects/CelestialBodies/Earth.hpp>
+#include <OpenSpaceToolkit/Physics/Time/Instant.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Derived/Angle.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Length.hpp>
+
+#include <Global.test.hpp>
+
+using ostk::core::types::String;
+using ostk::core::types::Real;
+using ostk::core::types::Shared;
+using ostk::core::types::Index;
+using ostk::core::types::Size;
+using ostk::core::ctnr::Table;
+using ostk::core::ctnr::Array;
+using ostk::core::fs::File;
+using ostk::core::fs::Path;
+
+using ostk::physics::units::Length;
+using ostk::physics::units::Angle;
+using ostk::physics::time::Instant;
+using ostk::physics::time::DateTime;
+using ostk::physics::time::Scale;
+using ostk::physics::coord::Position;
+using ostk::physics::coord::spherical::LLA;
+using ostk::physics::coord::Frame;
+using ostk::physics::environment::atmospheric::earth::NRLMSISE00;
+using ostk::physics::env::ephem::SPICE;
+using EarthGravitationalModel = ostk::physics::environment::gravitational::Earth;
+
+// Expose protected members for testing
+class NRLMSISE00Public : public NRLMSISE00
```

Comment:
You could maybe use a test fixture + friend class to do the same https://github.com/google/googletest/blob/main/docs/advanced.md#testing-private-code

Reply:
I did see that the `FRIEND_TEST` macro exists, but I kinda avoided using it because I _think_ having that macro in the header means requiring `gtest.h` to be a production dependency of OSTk, which seems a little gross.

I'm super open to sugestions here so let me know if i'm missing something with this.